### PR TITLE
GROK-990 Function have Javascript code-gen

### PR DESCRIPTION
**Pull request by jessica-norris**
This PR adds Javascript code-gen for existing Generic Function blocks (new style), needed for the Cookie clicker Blockly course coming up.

It also adds new copies of the definition blocks that don't nest (no next or previous statement connectors) (requested by the ACA). These new blocks don't have Javascript code-gen (they are Python only, at this stage).

### CODE REVIEW

File: generators/javascript/functions.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  Blockly.JavaScript[CALL_BLOCKS[i]] = Blockly.JavaScript['functions_call'];
+}
+
+/**
+ * Returns a value (from a function).
+ * @param {!Blockly.Block} block
+ * @returns {!Array|string}
+ */
+Blockly.JavaScript['functions_return'] = function(block) {
+  var value = Blockly.JavaScript.valueToCode(block, 'VALUE', Blockly.JavaScript.ORDER_NONE) || 'None';
```
Comment (timdawborn) [inline generators/javascript/functions.js]: `None` isn't valid JavaScript.

File: generators/javascript/functions.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+ * @param {!Blockly.Block} block
+ * @returns {!Array|string}
+ */
+Blockly.JavaScript['functions_call'] = function(block) {
+  // Grabs the real name of the function, based on the display name.
+  var functionName = Blockly.JavaScript.variableDB_.getName(
+      block.getFieldValue('NAME'), Blockly.Procedures.NAME_TYPE);
+  var params = [];
+  for (var i=0; i < block.getNumParams(); i++) {
+    params.push(Blockly.JavaScript.valueToCode(block, 'ARG_' + i, Blockly.JavaScript.ORDER_NONE) || 'None');
```
Comment (timdawborn) [inline generators/javascript/functions.js]: `None` isn't valid JavaScript.

File: generators/javascript/functions.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+ * @param {!Blockly.Block} block
+ * @returns {!Array|string}
+ */
+Blockly.JavaScript['functions_def'] = function(block) {
+  var funcName = Blockly.JavaScript.variableDB_.getName(
+      block.getFieldValue('NAME'), Blockly.Procedures.NAME_TYPE);
+  var branch = Blockly.JavaScript.statementToCode(block, 'STACK');
+  // Add a 'pass' if the contents are empty.
+  if (!branch) {
+    branch = Blockly.JavaScript.PASS;
```
Comment (timdawborn) [inline generators/javascript/functions.js]: From what I can see, `Blockly.JavaScript.PASS` is undefined here and in `blockly`?

**Comment by jessica-norris** [comment]
Okay, @timdawborn how about now? In practice we won't ever have an empty branch inside the function (blocks must not be empty in order to generate code) and there will always be the correct number of arguments (unless someone really fiddles with the block?) so I left a blank function body (since we have squiggly braces) and made the 'None' be 'null' instead.

**Pull request merged**